### PR TITLE
React dev doc dark theme

### PR DESCRIPTION
**Pull request by huayunh**
There is no JIRA issue associated with this.

## Changes proposed in this Pull Request:

- Added a theme toggler
- Ensured that most of the UI work in the dark theme

## Screenshots / Screen Recording (if applicable)

image

## To Test

- [Toggle light/dark theme in Chrome's developer tool](https://stackoverflow.com/questions/57606960/how-can-i-emulate-prefers-color-scheme-media-query-in-chrome)
- Known issue: when switching between light and dark theme, the underline in a tab always gets misaligned, like this: 
image
- Known issue: https://github.com/etn-ccis/blui-react-themes/issues/66

### CODE REVIEW

File: docs/src/shared/components/InfoCard/InfoCard.tsx [hunk 1]
```diff
@@ -97,7 +97,9 @@ export const InfoCard: React.FC<InfoCardProps> = (props): JSX.Element => {
                     </div>
                 </Box>
             )}
-            <Typography variant={'h6'}>{props.title}</Typography>
+            <Typography variant={'h6'} color={'text.primary'}>
+                {props.title}
+            </Typography>
             <Typography variant={'body2'} style={{ color: theme.palette.text.secondary, marginTop: theme.spacing(1) }}>
```
Comment (surajeaton) [inline docs/src/shared/components/InfoCard/InfoCard.tsx:103]: I missed this while migrating from doc-it. Can you please update this to have sx prop:
`sx={{color: 'text.secondary', mt: 1}}`

**Review by surajeaton — COMMENTED** [review]
I see the different styles in the react icon at left top drawer header while in the light and dark themes:

https://user-images.githubusercontent.com/120575281/230007104-d6793746-ece7-461c-9120-04f2cd0713c4.mov

**Comment by huayunh** [comment]
Thank you, @surajeaton . I have addressed all the concerns you've raised. The chips shifting around is a react theme thing — I have logged an issue [here](https://github.com/etn-ccis/blui-react-themes/issues/66). Feel free to take another pass at this.

**Comment by JeffGreiner-eaton** [comment]
in dark theme the mui-backdrop is being applied when it is set to invisible. Might be a zIndex thing

image

**Comment by JeffGreiner-eaton** [comment]
are there plans for system default or can it br removed?
image

**Comment by JeffGreiner-eaton** [comment]
playground scoreCard looks odd.
image

**Comment by huayunh** [comment]
> are there plans for system default or can it be removed?

"system default" is whatever your browser theme is set to be. And your browser is most likely following the preference setting of your operating system. So if your macOS is set to be the dark theme, with "system default" selected, you should see the dark theme used in the application.

**Comment by huayunh** [comment]
> playground scoreCard looks odd.

There are quite a few components with strange playground dark theme colors. These changes are better addressed at the component level — for example, by allowing users to pass `"background.paper"` to a `backgroundColor` prop. Therefore I left them the way they are.

**Comment by huayunh** [comment]
I am totally open to new naming suggestions on that.

**Review by surajeaton — APPROVED** [review]
Good job @huayunh.

**Pull request merged**